### DISCUSSION TITLE
App starts on Linux but not Windows?

### DISCUSSION THREAD

**Discussion by mysteryx93** [AvaloniaUI/Avalonia#10301] [Q&A]
I've updated the [MvvmDialogs samples](https://github.com/mysteryx93/HanumanInstitute.MvvmDialogs/) and am nearly done.

Something weird though... Avalonia demos work perfectly fine on Linux, but when I run on Windows (in a VM), they end silently upon starting. Scratching my head.

Tried with a blank template. That one runs. But the cross-platform sample in my project doesn't run.

Any idea?

Avalonia11-preview5
OS: Windows 10 in VM

and only other problem I'm having is that FluentAvalonia TaskDialogs size is completely broken, windows are way too small; but that's a totally separate issue. Then I should *finally* be able to deploy a MvvmDialogs for Avalonia11 and mobile devices.

**Comment by maxkatz6** [comment]
If it's crashing on windows VM, can you check Event Viewer logs? There usually is something useful.
Also, make sure app was published with `win7-x64` (in .NET 7 it's win-x64 by default, which might cause some problems, but it shouldn't crash anyway). With latest nightly builds win7-x64 isn't strongly required though.

**Reply by mysteryx93** [reply]
in this case app isn't published, I just hit Run or Debug

**Reply by maxkatz6** [reply]
Still, there should be something in the event viewer

**Reply by mysteryx93** [reply]
```
Faulting application name: Demo.Avalonia.ActivateNonModalDialog.exe, version: 1.0.0.0, time stamp: 0x638f99ee
Faulting module name: KERNELBASE.dll, version: 10.0.19041.2364, time stamp: 0x5b7d4d22
Exception code: 0xe0434352
Fault offset: 0x000000000002cd29
Faulting process id: 0x1af0
Faulting application start time: 0x01d93daff2673d56
Faulting application path: H:\GitHub\HanumanInstitute.MvvmDialogs\samples\Avalonia\Demo.ActivateNonModalDialog\bin\Debug\net7.0\Demo.Avalonia.ActivateNonModalDialog.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 5e5d3124-eb31-4e62-a227-1f205dd7812c
Faulting package full name: 
Faulting package-relative application ID: 
```
Not super useful.

**Reply by maxkatz6** [reply]
Maybe it will be possible to intercept any exception in the Main method with try/catch?

**Reply by mysteryx93** [reply]
huh... the Main method does not get called at all??

**Comment by mysteryx93** [comment]
OK. If I set this on the project, it launches fine.

```xml
<TargetFramework>net7.0-windows</TargetFramework>
```

With this, the Main method does not get called on Windows.

```xml
<TargetFramework>net7.0</TargetFramework>
```

I closed VIsual Studio and rebooted the VM a few times.
